src/BuildScriptGenerator/Node/NodePlatform.cs

```diff
@@ -125,6 +140,29 @@ public BuildScriptSnippet GenerateBashBuildScriptSnippet(BuildScriptGeneratorCon
             }
 
             bool pruneDevDependencies = ShouldPruneDevDependencies(context);
+            bool appInsightDependency = IsPackageDependencyExist(packageJson, NodeConstants.NodeAppInsightsPackageName);
+            string appInsightsInjectCommand = string.Empty;
+
+            if (_environment != null)
+            {
+                var appInsightsKey = _environment.GetEnvironmentVariable(Constants.AppInsightsKey);
+
+                // node_options is only supported in version 8.0.0 or newer and in 6.12.0
+                // so we will be able to set up app-insight only when node version is 6.12.0 or 8.0.0 or newer
+                if (!appInsightDependency
+                    && !string.IsNullOrEmpty(appInsightsKey)
+                    && (SemanticVersionResolver.CompareVersions(context.NodeVersion, "8.0.0") >= 0
+                    || SemanticVersionResolver.CompareVersions(context.NodeVersion, "6.12.0") == 0
+                    || SemanticVersionResolver.CompareVersions(context.LanguageVersion, "8.0.0") >= 0
+                    || SemanticVersionResolver.CompareVersions(context.LanguageVersion, "6.12.0") == 0))
+                {
+                    appInsightsInjectCommand =
+                        string.Concat(NodeConstants.NpmPackageInstallCommand, " --save ", NodeConstants.NodeAppInsightsPackageName);
```

Comment:
@gpcastro  .. I can see your point of view, but if we don't save it, next time when user trigger a code change, this will need to be executed, do we prefer to do that?